tests/integration/test_npm.py

```diff
@@ -262,6 +262,26 @@ def test_npm_smoketest(
             ],
             id="npm_lockfile3_aliased_deps",
         ),
+        pytest.param(
+            utils.TestParameters(
+                repo="https://github.com/cachito-testing/npm-dev-optional-peer-deps.git",
+                ref="d776f19ab6808dcb1c8e12e3d71322cb664957ce",
+                packages=({"path": ".", "type": "npm"},),
+                check_vendor_checksums=False,
+            ),
+            ["cat", "/opt/npm-ls-output.txt"],
+            [
+                "/tmp/npm_lockfile3_dev_optional_peer_deps-source/node_modules/ansi-styles:ansi-styles@6.2.1",
+                "/tmp/npm_lockfile3_dev_optional_peer_deps-source/node_modules/chalk:chalk@5.3.0",
+                "/tmp/npm_lockfile3_dev_optional_peer_deps-source/node_modules/emoji-regex:emoji-regex@10.2.1",
+                "/tmp/npm_lockfile3_dev_optional_peer_deps-source/node_modules/foo:foo@1.0.0:/tmp/npm_lockfile3_dev_optional_peer_deps-source/foo",
+                "/tmp/npm_lockfile3_dev_optional_peer_deps-source/node_modules/jsonfile:jsonfile@6.1.0",
+                "/tmp/npm_lockfile3_dev_optional_peer_deps-source/node_modules/minimist:minimist@1.2.8",
+                "/tmp/npm_lockfile3_dev_optional_peer_deps-source/node_modules/ansi-regex:ansi-regex@6.0.1",
+                "/tmp/npm_lockfile3_dev_optional_peer_deps-source/node_modules/balanced-match:balanced-match@2.0.0",
```

Comment:
I noticed that there are two transitive dependencies missing from this `npm ll` output: graceful-fs and universalify. I wonder if this only list direct dependencies :thinking: 

Will investigate :detective:

Reply:
This is `npm ll` aka `npm ls` fault. We need to add a `-a` to list transitive deps.

Reply:
Just noting that the `npm ls` command doesn't report the installed package version without the `-l` option. Since this test doesn't deal with multiple versions of the same package, I'm not sure that it matters though